### PR TITLE
refactor: change warning icon to use icons

### PR DESCRIPTION
**Pull request by pkdotson**
### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
before 
Screen Shot 2021-05-04 at 8 38 29 AM

after
Screen Shot 2021-05-04 at 8 38 43 AM

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/CRUD/data/dataset/AddDatasetModal.tsx [hunk 1]
```diff
@@ -107,7 +109,7 @@ const DatasetModal: FunctionComponent<DatasetModalProps> = ({
       show={show}
       title={
         <>
-          <StyledIcon name="warning-solid" />
+          <StyledIcon iconColor={theme.colors.grayscale.base} />
```
Comment (geido) [inline superset-frontend/src/views/CRUD/data/dataset/AddDatasetModal.tsx:112]: Hey @pkdotson same comment as the other PR. What's making defining the color specifically for the icon necessary here? The icon should get the color from the container

Comment (pkdotson) [reply superset-frontend/src/views/CRUD/data/dataset/AddDatasetModal.tsx:112]: oh I see. I'll redo this!

Comment (rusackas) [reply superset-frontend/src/views/CRUD/data/dataset/AddDatasetModal.tsx:112]: @pkdotson is there follow-up work to do on this?

Comment (junlincc) [reply superset-frontend/src/views/CRUD/data/dataset/AddDatasetModal.tsx:112]: @pkdotson bump ^

Comment (geido) [reply superset-frontend/src/views/CRUD/data/dataset/AddDatasetModal.tsx:112]: @rusackas @pkdotson see my comment here https://github.com/apache/superset/pull/14463#discussion_r635963140. I believe this falls under the same scenario.

Comment (pkdotson) [reply superset-frontend/src/views/CRUD/data/dataset/AddDatasetModal.tsx:112]: Going to keep it color prop for not until we can come up with default color for icon

**Comment by pkdotson** [comment]
Going to close this as there is no longer an icon for add dataset modal

**Pull request closed without merging**